lib/github-pages/version_check.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require "typhoeus"
+
+module GitHubPages
+  # VersionCheck ensures that the local version is up to date with the
+  # latest version deployed to GitHub Pages.
+  module VersionCheck
+    PRODUCTION_VERSIONS_URL = "https://pages.github.com/versions.json"
+
+    def run!
+      @already_run ||=
+        begin
+          return true if ENV["PAGES_ENV"].to_s.casecmp("enterprise").zero?
+
+          if GitHubPages::VERSION < production_ghp_version
+            Jekyll.logger.error "GitHub Pages:", "You are using an outdated version of github-pages."
+            Jekyll.logger.error "", "Upgrade to version '#{production_ghp_version}' to ensure your local builds"
+            Jekyll.logger.error "", "will accurately reflect what is deployed when you push to GitHub."
+            return false
```

Comment:
Agreed. This link? https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/#keeping-your-site-up-to-date-with-the-github-pages-gem